diff -u -r openthread-master/configure.ac openthread-modified/configure.ac
--- openthread-master/configure.ac     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/configure.ac     2017-06-13 19:44:01.178447178 +0530
@@ -1048,6 +1048,36 @@
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_RAW_LINK_API],[${OPENTHREAD_ENABLE_RAW_LINK_API}],[Define to 1 if you want to enable raw link-layer API])

 #
+# Added to enable switching between nordic radio (IEEE802.15.4-2006) and  DW1000 Radio (IEEE802.15.4-2011)
+#
+AC_ARG_ENABLE(dw1000,
+    [AS_HELP_STRING([--enable-dw1000],[Enable DW1000 radio support @<:@default=no@:>@.])],
+    [
+        case "${enableval}" in
+
+        no|yes)
+            enable_dw1000=${enableval}
+            ;;
+
+        *)
+            AC_MSG_ERROR([Invalid value ${enable_dw1000} for --enable-dw1000])
+            ;;
+        esac
+    ],
+    [enable_dw1000=no])
+
+if test "$enable_dw1000" = "yes"; then
+        OPENTHREAD_ENABLE_DW1000_RADIO=1
+else
+        OPENTHREAD_ENABLE_DW1000_RADIO=0
+fi
+
+AC_MSG_RESULT(${enable_dw1000})
+AC_SUBST(OPENTHREAD_ENABLE_DW1000_RADIO)
+AM_CONDITIONAL([OPENTHREAD_ENABLE_DW1000_RADIO], [test "${enable_dw1000}" = "yes"])
+AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_DW1000_RADIO], [${OPENTHREAD_ENABLE_DW1000_RADIO}],[Define to 1 if you want to use dw1000 radio])
+
+#
 # Examples
 #

diff -u -r openthread-master/examples/common-switches.mk openthread-modified/examples/common-switches.mk
--- openthread-master/examples/common-switches.mk     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/examples/common-switches.mk     2017-06-13 19:44:01.158447178 +0530
@@ -85,3 +85,7 @@
 ifeq ($(COAP),1)
 configure_OPTIONS              += --enable-application-coap
 endif
+
+ifeq ($(ENABLE_DW1000),1)
+configure_OPTIONS              += --enable-dw1000
+endif
diff -u -r openthread-master/examples/platforms/nrf52840/Makefile.am openthread-modified/examples/platforms/nrf52840/Makefile.am
--- openthread-master/examples/platforms/nrf52840/Makefile.am     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/examples/platforms/nrf52840/Makefile.am     2017-06-13 19:44:01.158447178 +0530
@@ -49,19 +49,36 @@
     -I$(top_srcdir)/third_party/NordicSemiconductor/segger_rtt                   \
     $(NULL)

+if OPENTHREAD_ENABLE_DW1000_RADIO
+COMMONCPPFLAGS                                                                += \
+     -I$(top_srcdir)/third_party/decawave/dw1000/common                          \
+     -I$(top_srcdir)/third_party/decawave/dw1000/drivers                         \
+     -I$(top_srcdir)/third_party/decawave/dw1000/device                          \
+    $(NULL)
+endif
+
 PLATFORM_SOURCES                                                               = \
     alarm.c                                                                      \
     flash.c                                                                      \
     logging.c                                                                    \
     misc.c                                                                       \
-    platform.c                                                                   \
-    radio.c                                                                      \
+    platform.c                                                                   \                                                           \
     random.c                                                                     \
     uart.c                                                                       \
     platform-config.h                                                            \
     platform-nrf5.h                                                              \
     $(NULL)

+if  OPENTHREAD_ENABLE_DW1000_RADIO
+PLATFORM_SOURCES                                                              += \
+   ../dw1000/dw1000_radio.c                                                      \
+    $(NULL)
+else
+PLATFORM_SOURCES                                                              += \
+    radio.c                                                                      \
+    $(NULL)
+endif
+
 if OPENTHREAD_ENABLE_DIAG
 PLATFORM_SOURCES                                                              += \
     diag.c                                                                       \
@@ -71,12 +88,26 @@
 NORDICSEMI_SOURCES                                                             = \
     @top_builddir@/third_party/NordicSemiconductor/hal/nrf_nvmc.c                \
     @top_builddir@/third_party/NordicSemiconductor/device/system_nrf52840.c      \
-    @top_builddir@/third_party/NordicSemiconductor/drivers/nrf_drv_radio802154.c \
     @top_builddir@/third_party/NordicSemiconductor/segger_rtt/SEGGER_RTT.c       \
     @top_builddir@/third_party/NordicSemiconductor/segger_rtt/SEGGER_RTT_Conf.h  \
     @top_builddir@/third_party/NordicSemiconductor/device/gcc_startup_nrf52840.S \
     $(NULL)

+if  OPENTHREAD_ENABLE_DW1000_RADIO
+NORDICSEMI_SOURCES                                                            += \
+    @top_builddir@/third_party/decawave/dw1000/common/dw1000_driver.c            \
+    @top_builddir@/third_party/decawave/dw1000/drivers/dw1000_device.c           \
+    @top_builddir@/third_party/decawave/dw1000/device/dw1000_spi.c               \
+    @top_builddir@/third_party/decawave/dw1000/device/dw1000_mutex.c             \
+    @top_builddir@/third_party/decawave/dw1000/drivers/dw1000_params_init.c      \
+    @top_builddir@/third_party/NordicSemiconductor/drivers/nrf_drv_spi.c         \
+    $(NULL)
+else
+NORDICSEMI_SOURCES                                                            += \
+    @top_builddir@/third_party/NordicSemiconductor/drivers/nrf_drv_radio802154.c \
+    $(NULL)
+endif
+
 libopenthread_nrf52840_a_CPPFLAGS                                              = \
     $(COMMONCPPFLAGS)                                                            \
     $(NULL)
@@ -114,7 +145,6 @@
     $(top_srcdir)/third_party/NordicSemiconductor/device/nrf52840_peripherals.h  \
     $(top_srcdir)/third_party/NordicSemiconductor/device/system_nrf52840.h       \
     $(top_srcdir)/third_party/NordicSemiconductor/drivers/nrf_drv_clock.h        \
-    $(top_srcdir)/third_party/NordicSemiconductor/drivers/nrf_drv_radio802154.h  \
     $(top_srcdir)/third_party/NordicSemiconductor/hal/nrf_assert.h               \
     $(top_srcdir)/third_party/NordicSemiconductor/hal/nrf_clock.h                \
     $(top_srcdir)/third_party/NordicSemiconductor/hal/nrf_egu.h                  \
@@ -126,6 +156,19 @@
     $(top_srcdir)/third_party/NordicSemiconductor/segger_rtt/SEGGER_RTT.h        \
     $(NULL)

+if  OPENTHREAD_ENABLE_DW1000_RADIO
+noinst_HEADERS                                                               +=  \
+    $(top_srcdir)/third_party/decawave/dw1000/common/dw1000_driver.h             \
+    $(top_srcdir)/third_party/decawave/dw1000/drivers/dw1000_device_api.h        \
+    $(top_srcdir)/third_party/decawave/dw1000/device/dw1000_spi.h                \
+    $(top_srcdir)/third_party/NordicSemiconductor/drivers/nrf_drv_spi.h          \
+    $(NULL)
+else
+noinst_HEADERS                                                                += \
+    $(top_srcdir)/third_party/NordicSemiconductor/drivers/nrf_drv_radio802154.h  \
+    $(NULL)
+endif
+
 PRETTY_FILES                                                                   = \
     $(PLATFORM_SOURCES)                                                          \
     $(NULL)
diff -u -r openthread-master/examples/platforms/nrf52840/platform.c openthread-modified/examples/platforms/nrf52840/platform.c
--- openthread-master/examples/platforms/nrf52840/platform.c     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/examples/platforms/nrf52840/platform.c     2017-06-13 19:44:01.158447178 +0530
@@ -40,6 +40,10 @@
 #include <openthread-core-config.h>
 #include <openthread-config.h>

+#if OPENTHREAD_ENABLE_DW1000_RADIO
+#include "dw1000/dw1000_radio.h"
+#endif
+
 void __cxa_pure_virtual(void) { while (1); }

 void PlatformInit(int argc, char *argv[])
@@ -53,7 +57,13 @@
     nrf5UartInit();
     nrf5MiscInit();
     nrf5CryptoInit();
+
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    dw1000RadioInit();
+#else
     nrf5RadioInit();
+#endif
+
 #if (OPENTHREAD_ENABLE_DEFAULT_LOGGING == 0)
     nrf5LogInit();
 #endif
@@ -61,10 +71,16 @@

 void PlatformDeinit(void)
 {
+
 #if (OPENTHREAD_ENABLE_DEFAULT_LOGGING == 0)
     nrf5LogDeinit();
 #endif
+
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+#else
     nrf5RadioDeinit();
+#endif
+
     nrf5CryptoDeinit();
     nrf5MiscDeinit();
     nrf5UartDeinit();
@@ -75,6 +91,11 @@
 void PlatformProcessDrivers(otInstance *aInstance)
 {
     nrf5AlarmProcess(aInstance);
+
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    dw1000RadioProcess(aInstance);
+#else
     nrf5RadioProcess(aInstance);
+#endif
     nrf5UartProcess();
 }
diff -u -r openthread-master/include/openthread/coap.h openthread-modified/include/openthread/coap.h
--- openthread-master/include/openthread/coap.h     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/include/openthread/coap.h     2017-06-16 14:07:57.173977000 +0530
@@ -508,6 +508,32 @@
  *
  */
 otError otCoapSendResponse(otInstance *aInstance, otMessage *aMessage, const otMessageInfo *aMessageInfo);
+
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+
+/**
+ * This function will send the widget value into the cloud.
+ *
+ * @param[in]  aInstance        A pointer to an OpenThread instance.
+ * @param[in]  aUriPath         Cloud URI path
+ * @param[in]  aPayload         Payload which is to be sent to the cloud.
+ *
+ */
+void otCloudDataSend(otInstance * aInstance,
+                     const char * aUriPath,
+                     char       * aPayload);
+
+/**
+ * This function will create the payload and Updates the integer value
+ * to the cloud.
+ *
+ * @param[in]  aInstance        A pointer to an OpenThread instance.
+ * @param[in]  aWidgetVal       The value which will be sent to the cloud.
+ *
+ */
+void otCloudDataUpdate(otInstance * aInstance, uint16_t aWidgetVal);
+
+#endif
 #endif // OPENTHREAD_ENABLE_APPLICATION_COAP

 /**
diff -u -r openthread-master/include/openthread/platform/radio.h openthread-modified/include/openthread/platform/radio.h
--- openthread-master/include/openthread/platform/radio.h     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/include/openthread/platform/radio.h     2017-06-14 20:21:49.642935000 +0530
@@ -67,12 +67,21 @@
 enum
 {
     OT_RADIO_FRAME_MAX_SIZE     = 127,                             ///< aMaxPHYPacketSize (IEEE 802.15.4-2006)
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    OT_RADIO_CHANNEL_MIN         = 5,
+    OT_RADIO_CHANNEL_MAX         = 5,
+#else
     OT_RADIO_CHANNEL_MIN         = 11,                              ///< 2.4 GHz IEEE 802.15.4-2006
     OT_RADIO_CHANNEL_MAX         = 26,                              ///< 2.4 GHz IEEE 802.15.4-2006
+#endif
+
     OT_RADIO_SUPPORTED_CHANNELS  = 0xffff << OT_RADIO_CHANNEL_MIN,  ///< 2.4 GHz IEEE 802.15.4-2006
     OT_RADIO_SYMBOLS_PER_OCTET   = 2,                               ///< 2.4 GHz IEEE 802.15.4-2006
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    OT_RADIO_BIT_RATE            = 6800000,
+#else
     OT_RADIO_BIT_RATE            = 250000,                          ///< 2.4 GHz IEEE 802.15.4 (kilobits per second)
-
+#endif
     OT_RADIO_BITS_PER_OCTET  = 8,      ///< Number of bits per octet
     OT_RADIO_SYMBOL_TIME     = ((OT_RADIO_BITS_PER_OCTET / OT_RADIO_SYMBOLS_PER_OCTET) * 1000000) / OT_RADIO_BIT_RATE,

@@ -84,6 +93,7 @@
  * This enum represents radio capabilities.
  *
  */
+
 typedef enum otRadioCaps
 {
     OT_RADIO_CAPS_NONE              = 0,  ///< None
@@ -159,6 +169,15 @@
 void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64);

 /**
+ * Get the Radio PanId.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ *
+ * @returns the Radio PanId
+ */
+uint16_t otPlatformGetPanId(otInstance *aInstance);
+
+/**
  * Set the PAN ID for address filtering.
  *
  * @param[in] aInstance  The OpenThread instance structure.
@@ -523,6 +542,16 @@
 int8_t otPlatRadioGetReceiveSensitivity(otInstance *aInstance);

 /**
+ * The Cli command "logprint" calls this method to Print any variable or data.
+ *
+ * @param[out]  buffer is the value to be printed.
+ *
+ * @returns the size of the buffer
+ */
+uint8_t otPlatRadioPrintBuf(uint8_t *buffer);
+
+
+/**
  * @}
  *
  */
diff -u -r openthread-master/src/cli/cli.cpp openthread-modified/src/cli/cli.cpp
--- openthread-master/src/cli/cli.cpp     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/src/cli/cli.cpp     2017-06-16 14:12:19.545972000 +0530
@@ -74,10 +74,16 @@

 #if OPENTHREAD_ENABLE_APPLICATION_COAP
 #include "cli_coap.hpp"
+#include "openthread/coap.h"
 #endif

 #include "common/encoding.hpp"

+#if OPENTHREAD_ENABLE_DW1000_RADIO
+#include <openthread/platform/radio.h>
+#endif
+
+
 using ot::Encoding::BigEndian::HostSwap16;
 using ot::Encoding::BigEndian::HostSwap32;

@@ -99,6 +105,9 @@
     { "childtimeout", &Interpreter::ProcessChildTimeout },
 #if OPENTHREAD_ENABLE_APPLICATION_COAP
     { "coap", &Interpreter::ProcessCoap },
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    { "coapsend", &Interpreter::ProcessCoapSend},
+#endif
 #endif
 #if OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD
     { "commissioner", &Interpreter::ProcessCommissioner },
@@ -192,6 +201,9 @@
     { "txpowermax", &Interpreter::ProcessTxPowerMax },
     { "version", &Interpreter::ProcessVersion },
     { "whitelist", &Interpreter::ProcessWhitelist },
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    { "logprint", &Interpreter::ProcessLogPrint},
+#endif
 };

 #ifdef OTDLL
@@ -3124,5 +3136,39 @@
 }
 #endif

+#if OPENTHREAD_ENABLE_DW1000_RADIO
+
+void Interpreter::ProcessLogPrint(int argc, char *argv[])
+{
+    (void)argc;
+    (void)argv;
+    uint8_t buffer[32];
+    uint8_t size = 0;
+
+    size = otPlatRadioPrintBuf(buffer);
+    OutputBytes(buffer, size);
+    sServer->OutputFormat("\r\n");
+}
+
+#if OPENTHREAD_ENABLE_APPLICATION_COAP
+
+void Interpreter::ProcessCoapSend(int argc, char *argv[])
+{
+    long temp= 0;
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(argc > 0, error = OT_ERROR_INVALID_ARGS);
+    SuccessOrExit(error = ParseLong(argv[0], (temp)));
+
+    otCloudDataUpdate(mInstance, static_cast<uint16_t>(temp));
+
+    sServer->OutputFormat("cloud value updated %d\r\n", temp);
+
+exit:
+    AppendResult(error);
+}
+
+#endif // OPENTHREAD_ENABLE_APPLICATION_COAP
+#endif // OPENTHREAD_ENABLE_DW1000_RADIO
 }  // namespace Cli
 }  // namespace ot
diff -u -r openthread-master/src/cli/cli.hpp openthread-modified/src/cli/cli.hpp
--- openthread-master/src/cli/cli.hpp     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/src/cli/cli.hpp     2017-06-16 14:14:52.545969000 +0530
@@ -170,6 +170,9 @@
 #endif
     void ProcessChildTimeout(int argc, char *argv[]);
 #if OPENTHREAD_ENABLE_APPLICATION_COAP
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    void ProcessCoapSend(int argc,char *argv[]);
+#endif
     void ProcessCoap(int argc, char *argv[]);
 #endif  //OPENTHREAD_ENABLE_APPLICATION_COAP
 #if OPENTHREAD_ENABLE_COMMISSIONER && OPENTHREAD_FTD
@@ -266,7 +269,9 @@
     void ProcessTxPowerMax(int argc, char *argv[]);
     void ProcessVersion(int argc, char *argv[]);
     void ProcessWhitelist(int argc, char *argv[]);
-
+#if OPENTHREAD_ENABLE_DW1000_RADIO
+    void ProcessLogPrint(int argc, char *argv[]);
+#endif
 #ifdef OTDLL
     void ProcessInstanceList(int argc, char *argv[]);
     void ProcessInstance(int argc, char *argv[]);
diff -u -r openthread-master/src/core/api/coap_api.cpp openthread-modified/src/core/api/coap_api.cpp
--- openthread-master/src/core/api/coap_api.cpp     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/src/core/api/coap_api.cpp     2017-06-16 14:17:40.617966000 +0530
@@ -38,6 +38,30 @@

 #if OPENTHREAD_ENABLE_APPLICATION_COAP

+#if OPENTHREAD_ENABLE_DW1000_RADIO
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <openthread/openthread.h>
+#include <openthread/diag.h>
+#include <openthread/cli.h>
+#include <openthread/platform/platform.h>
+#include <openthread/platform/alarm.h>
+
+/**< NAT64 address of the thethings.io cloud. */
+#define CLOUD_NAT64_ADDRESS           "fd00:64:123:4567::68c7:55d3"
+
+/**< Put your things URI here. */
+#define CLOUD_URI_PATH                "v2/things/oixa-fWQV2rPAB7oj5bEju7vlaafEpitg0Gogxz9PeM"
+
+/**< Thing resource name. */
+#define CLOUD_THING_RESOURCE          "Nordic-Decawave"
+
+/**< Use application/json content format type. */
+#define CLOUD_COAP_CONTENT_FORMAT     50
+#endif
+
 using namespace ot;

 void otCoapHeaderInit(otCoapHeader *aHeader, otCoapType aType, otCoapCode aCode)
@@ -180,4 +204,65 @@
                *static_cast<Message *>(aMessage), *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
 }

+#if OPENTHREAD_ENABLE_DW1000_RADIO
+void otCloudDataSend(otInstance * aInstance,
+                     const char * aUriPath,
+                     char       * aPayload)
+{
+     otError error = OT_ERROR_NO_BUFS;
+     otCoapHeader header;
+     otCoapOption content_format_option;
+     otMessage * p_request;
+     otMessageInfo message_info;
+     uint8_t content_format = CLOUD_COAP_CONTENT_FORMAT;
+
+     do
+     {
+         content_format_option.mNumber = kCoapOptionContentFormat;
+         content_format_option.mLength = 1;
+         content_format_option.mValue  = &content_format;
+
+         otCoapHeaderInit(&header, kCoapTypeNonConfirmable, kCoapRequestPost);
+         otCoapHeaderAppendUriPathOptions(&header, aUriPath);
+         otCoapHeaderAppendOption(&header, &content_format_option);
+         otCoapHeaderSetPayloadMarker(&header);
+
+         p_request = otCoapNewMessage(aInstance, &header);
+         if (p_request == NULL)
+         {
+             break;
+         }
+
+         error = otMessageAppend(p_request, aPayload, strlen(aPayload));
+         if (error != OT_ERROR_NONE)
+         {
+             break;
+         }
+
+         memset(&message_info, 0, sizeof(message_info));
+         message_info.mInterfaceId = OT_NETIF_INTERFACE_ID_THREAD;
+         message_info.mPeerPort = OT_DEFAULT_COAP_PORT;
+         otIp6AddressFromString(CLOUD_NAT64_ADDRESS, &message_info.mPeerAddr);
+
+         error = otCoapSendRequest(aInstance, p_request, &message_info, NULL, NULL);
+
+     } while (false);
+
+     if (error != OT_ERROR_NONE && p_request != NULL)
+     {
+         otMessageFree(p_request);
+     }
+ }
+
+void otCloudDataUpdate(otInstance * aInstance, uint16_t aWidgetVal)
+ {
+     char payload_buffer[64];
+
+     sprintf(payload_buffer,
+             "{\"values\":[{\"key\":\"%s\",\"value\":\"%d\"}]}",
+             CLOUD_THING_RESOURCE, aWidgetVal);
+
+     otCloudDataSend(aInstance, CLOUD_URI_PATH, payload_buffer);
+ }
+#endif // OPENTHREAD_ENABLE_DW1000_RADIO
 #endif // OPENTHREAD_ENABLE_APPLICATION_COAP
diff -u -r openthread-master/src/core/openthread-core-default-config.h openthread-modified/src/core/openthread-core-default-config.h
--- openthread-master/src/core/openthread-core-default-config.h     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/src/core/openthread-core-default-config.h     2017-06-13 19:44:01.058447179 +0530
@@ -102,7 +102,13 @@
  *
  */
 #ifndef OPENTHREAD_CONFIG_DEFAULT_CHANNEL
+
+#if  OPENTHREAD_ENABLE_DW1000_RADIO
+#define OPENTHREAD_CONFIG_DEFAULT_CHANNEL                       5
+#else
 #define OPENTHREAD_CONFIG_DEFAULT_CHANNEL                       11
+#endif
+
 #endif  // OPENTHREAD_CONFIG_DEFAULT_CHANNEL

 /**
diff -u -r openthread-master/third_party/mbedtls/repo/library/ssl_tls.c openthread-modified/third_party/mbedtls/repo/library/ssl_tls.c
--- openthread-master/third_party/mbedtls/repo/library/ssl_tls.c     2017-05-30 22:26:22.000000000 +0530
+++ openthread-modified/third_party/mbedtls/repo/library/ssl_tls.c     2017-06-13 19:44:01.094447178 +0530
